### PR TITLE
Fellowships directory changes

### LINKED ISSUE #1174: [Fellowship Mini Site] Add "In Residence" category to Directory pages
### Issue: We need a space for our "In Residence" fellows category in our Directory Pages.

**List of requirements:**  - Thanks @xmatthewx for suggesting it.
- [ ] Add new label `In Residence` (to the filter, to profiles)
- [ ] put in alphabetical order (issue #1175)
- [ ] link to section below with title `Fellows in Residence`
- [ ] include button to `See all Fellows in Residence`
- [ ] link button to page at `/fellowships/directory/in-residence/`
- [ ] page should include all profiles in this category
- [ ] **In Pulse:** Change program type on Pulse backend to `In Residence` (it's currently set to `Fellow in Residence`)

---------------------------
**Visual walkthrough:**

We should add "In Residence" to the filter: 

image

---------------------------

Also, to this page (as an area by itself):

image

------------------------

And also to its own page, like we have for other areas such as "Science":

image
----------------------

Here is the list of fellows in Residence for context (extracted from [here](https://github.com/mozilla/foundation.mozilla.org/issues/1163#issuecomment-371321049)):
Philip Smith - Misinformation + Fake News
Cori Zarek - Tech Policy
An-Me Chung - Web Literacy
Jon Rogers - IoT
Peter Bihr - IoT
Julia Kloiber - IoT
Steve Song - Access and Digital Inclusion
Renee DiResta (coming soon) - Misinformation
Thomas Lohninger- TBD (focus on enforcing Net Neutrality in the EU, but may be titled something else)

@mmmavis let me know if this is clear. In can you need a comp, I can quickly provide one.

cc: @xmatthewx - feel free to edit this issue to add more details if you want :)

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #1174, #1175 

- fellowships directory section are now in alphabetical order
- added 'in residence' section"

Things can be cleaned up / refactored further but I didn't wanna modify existing code drastically yet as specs seem to be somewhat in flux still...

see non whitespaces code changes: https://github.com/mozilla/foundation.mozilla.org/pull/1179/files?w=1

---

@cadecairos I've introduced 2 new env vars in this PR. I wanted to make sure I did it correctly. I've added them to 
- .travis.yml
- app.json
- appveyor.yml
-  env.default

Did I miss anything?

### CODE REVIEW

**Comment by mmmavis** [comment]
@taisdesouzalessa can you help check design/content? If things look okay I'll pass it to someone else for code review? https://foundation-mofostaging-pr-1179.herokuapp.com/fellowships/directory/

**Comment by taisdesouzalessa** [comment]
Everything looks great form the design standpoint, Mavis! 
There is only one language thing that seems off to me, we now have `Fellows, In Residence 2017` I wonder if `Fellows, In Residence, 2017` would be better because to me `Fellows, In Residence 2017` it reads that they were In residence only for 2017 but have always been a Fellow. English is my second language so I could be wrong. I'll defer to @xmatthewx who had saved me in so many writing issues before...lol. Matthew, can you give your opinion here?

File: source/js/fellowships.js [hunk 1]
```diff
@@ -6,10 +6,16 @@ import ReactDOM from 'react-dom';
...
 import LoadingIndicator from './components/loading-indicator/loading-indicator.jsx';
 
+const DIRECTORY_PAGE_FILTER_OPTIONS = {'program_year': `2017`};
+
 let pulseApiDomain = ``;
 let pulseDomain = ``;
-const DIRECTORY_PAGE_FILTER_OPTIONS = {'program_year': `2017`};
-const DIRECTORY_PAGE_TYPE_ORDER = [ `science`, `open web`, `tech policy`, `media`];
+
+let capitalize = function(str) {
```
Comment (gvn) [inline source/js/fellowships.js]: Could use arrow function here...

File: source/js/fellowships.js [hunk 2]
```diff
@@ -171,7 +172,7 @@ function injectReactComponents(env) {
 
   // Fellows on Fellows Directory page
   if (document.getElementById(`fellows-directory-featured-fellows`)) {
-    renderFellowsOnDirectoryPage();
+    renderFellowsOnDirectoryPage(env.FELLOWSHIPS_DIRECTORY_TYPES.slice(0));
```
Comment (mmmavis) [inline source/js/fellowships.js]: i wonder if we really need to create a new copy of `env.FELLOWSHIPS_DIRECTORY_TYPES` here?

**Comment by xmatthewx** [comment]
Good catch @taisdesouzalessa. @mmmavis let's add that comma.

File: source/js/fellowships.js (hunk 1 above)
Comment (mmmavis) [reply source/js/fellowships.js]: nice catch! will update

**Comment by mmmavis** [comment]
@cadecairos both code and design have been reviewed and okayed. Please R+ this PR if env vars are added correctly. Thanks!!! 😃

**Comment by alanmoo** [comment]
Stumbled through this trying to get reacquainted with everything: I haven't dug deep, but at first glance the way those env vars are being used it feels like they'd be much more appropriate as constants in the code, or perhaps we need changes to the Pulse API.

File: network-api/networkapi/settings.py [hunk 3]
```diff
@@ -490,6 +490,8 @@
     'NETWORK_SITE_URL': env('NETWORK_SITE_URL'),
     'TARGET_DOMAIN': env('TARGET_DOMAIN'),
     'SHOW_TAKEOVER': env('SHOW_TAKEOVER'),
+    'FELLOWSHIPS_PROGRAM_INFO_TYPES': env('FELLOWSHIPS_PROGRAM_INFO_TYPES').split(','),
```
Comment (cadecairos) [inline network-api/networkapi/settings.py]: If you tell environ about these vars, and declare them as `list` types, you can avoid having to manually split them.

Here's an example of how we load the ALLOWED_HOSTS variable: https://github.com/mozilla/foundation.mozilla.org/blob/c1a9b27e69ee6c80107cc8d326f982f900dea140/network-api/networkapi/settings.py#L24-L26

**Comment by mmmavis** [comment]
## DO MERGE YET ##

Seems like we want to change the directory pages drastically... so we might not end up needing this PR 😬 I'm gonna put this one on hold for now......

**Comment by mmmavis** [comment]
🛑 Team has decided to redesign the directory pages so this PR is not applicable anymore. I'm closing it now.

(For more context see [here](https://github.com/mozilla/foundation.mozilla.org/issues/1185))

**Pull request closed without merging**